### ISSUE TITLE
Mutual TLS (mTLS / SSL client authentication) Support

### ISSUE THREAD

**Issue by pxeger** [opnsense/core#4924]
**Important notices**

Before you add a new report, we ask you kindly to acknowledge the following:

- [x] I have read the contributing guide lines at https://github.com/opnsense/core/blob/master/CONTRIBUTING.md
- [x] I have searched the existing issues and I am convinced that mine is new.

**Is your feature request related to a problem? Please describe.**

Mutual TLS is a way to authenticate clients that connect to a server, as well as the other way around, which is useful for securing access to services rather than using a VPN.

**Describe the solution you like**

Some options in the System -> Settings -> Administration page to set up mutual TLS:
- mode: one of disabled (ignore/reject mTLS), optional (request clients to connect with mTLS, and verify their certificates, but allow them to access it if they don't provide a valid certificate), or required (valid certificate must be presented in order to access)
- CAs: a list of Certificate Authorities (to be chosen from the System -> Trust -> Authorities) that valid certificates must have been issued by

**Describe alternatives you considered**

You could install HAproxy or some other plugin to proxy the web interface and add mTLS there, but then you could either just bypass it by connecting to the web interface directly, unless you disable it, which would then make recovery management difficult if the proxy breaks.

**Additional context**

https://smallstep.com/hello-mtls is a useful overview of mTLS.

**Comment by AdSchellevis** [comment]
similar too https://github.com/opnsense/core/issues/1718 ?

**Comment by pxeger** [comment]
Yes - I only searched for "mutual TLS" not "client authentication", sorry.

**Comment by AdSchellevis** [comment]
no problem, I think the old ticket contains usable data, but I don't mind keeping this open for a while to wait for more interest.
